src/NServiceBus.Core/Routing/RoutingMappingSettings.cs

```diff
@@ -32,13 +31,13 @@ internal RoutingMappingSettings(SettingsHolder settings)
         public Publishers Publishers => GetOrCreate<Publishers>();
 
         /// <summary>
-        /// Sets a distribution strategy for a given subset of message types.
+        /// Sets a distribution strategy for a given endpoint.
         /// </summary>
+        /// <param name="endpointName">The name of the logical endpoint the given strategy should apply to.</param>
         /// <param name="distributionStrategy">The instance of a distribution strategy.</param>
-        /// <param name="typeMatchingRule">A predicate for determining the set of types.</param>
-        public void SetMessageDistributionStrategy(DistributionStrategy distributionStrategy, Func<Type, bool> typeMatchingRule)
+        public void SetMessageDistributionStrategy(string endpointName, DistributionStrategy distributionStrategy)
```

Comment:
In the prev version we had the `func` which allowed to register a strategy for any set of types. That said, I would assume for any given endpoint there would be less downstream endpoints than message types it sends. I think the per-endpoint API is OK because ultimately the distribution strategy that should be used is not dependent on the endpoint which runs it but on the endpoint which is the target of messages.

In short, I am :+1: for the current API.